modality-probe-cli/src/log/radius.rs

```diff
@@ -0,0 +1,407 @@
+use std::{
+    collections::{BTreeMap, HashSet},
+    convert::TryFrom,
+};
+
+use modality_probe::{LogicalClock, ProbeId};
+use modality_probe_collector_common::{LogEntryData, ReportLogEntry, SequenceNumber};
+
+use crate::{error::CmdError, give_up, hopefully, hopefully_ok};
+
+use super::SortedProbes;
+
+#[derive(PartialEq, Debug)]
+pub struct Radius {
+    distance: usize,
+    probe_id: ProbeId,
+    seq: SequenceNumber,
+    seq_index: u32,
+}
+
+impl TryFrom<&str> for Radius {
+    type Error = Box<CmdError>;
+    fn try_from(s: &str) -> Result<Self, Self::Error> {
+        let sections: Vec<&str> = s.split(',').collect();
+        if sections.len() != 2 {
+            give_up!(format!(
+                "{} is not a valid radius. It's missing either the coordinate or the distance",
+                s
+            ));
+        }
+        let coord = sections[0];
+        let c_sections: Vec<&str> = coord.split(':').collect();
+        if c_sections.len() != 4 {
+            give_up!(format!("{} is not a valid coordinate", coord));
+        }
+
+        let distance = hopefully!(
+            sections[1].parse::<usize>(),
+            "Unable to parse the given distance"
```

Comment:
Can we make the coordinate sub-component parser errors unique, so that we can tell the user which part of the supplied coordinate was problematic?

Reply:
Nice, I like it.